components/src/core/IconSwitch/IconSwitch.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useState } from 'react';
+import { StyleSheet, TouchableOpacity, ViewProps, View } from 'react-native';
+import Animated, {
+    interpolate,
+    useAnimatedStyle,
+    useSharedValue,
+    withTiming,
+    Easing,
+    Extrapolation,
+    interpolateColor,
+} from 'react-native-reanimated';
+import { Icon } from '../Icon/Icon';
+import { $DeepPartial } from '@callstack/react-theme-provider';
+import { MD3Theme, useTheme } from 'react-native-paper';
+
+export type IconSwitchProps = ViewProps & {
+    checkedIcon?: boolean;
+    disabledSwitch?: boolean;
+    color?: string;
```

Comment:
I cannot think of an example where `color` would be used by our users. Why would you want to change colors?